runtime/near-vm/engine/src/universal/engine.rs

```diff
@@ -565,16 +565,17 @@ impl UniversalEngineInner {
         // - data section body size
         // -- padding between data sections
         let page_size = rustix::param::page_size();
-        let total_len = round_up(
-            function_bodies.iter().fold(0, |acc, func| {
-                round_up(acc + function_allocation_size(*func), ARCH_FUNCTION_ALIGNMENT.into())
-            }) + executable_sections.iter().fold(0, |acc, exec| {
-                round_up(acc + exec.bytes.len(), ARCH_FUNCTION_ALIGNMENT.into())
-            }),
-            page_size,
-        ) + data_sections
-            .iter()
-            .fold(0, |acc, data| round_up(acc + data.bytes.len(), DATA_SECTION_ALIGNMENT.into()));
+        let total_len = 0;
+        let total_len = function_bodies.iter().fold(total_len, |acc, func| {
+            round_up(acc + function_allocation_size(*func), ARCH_FUNCTION_ALIGNMENT.into())
```

Comment:
I don’t think this is correct. Here you’re first adding the allocation and then rounding up to ARCH, while ARCH is the alignment for the allocation itself and not for the next allocation. In practice it’ll work because we only have a single alignment type before aligning to page and then switching to another alignment type, but I think the proper way of doing this (more future-proof should we add other alignments later on) would be:
```rust
round_up(acc, ARCH_FUNCTION_ALIGNMENT.into()) + function_allocation_size(*func)
```
```

Reply:
Hm, so this really depends on who does one considers the padding bytes to belong to.

In assembly due to how labels work the padding bytes will usually appear as-if they belong to the preceding function:

```asm
func1:
  add
  div
  ret
  ud2 ; padding
  ud2 ; padding
func2:
  ...
```

Which sort-of matches the current code a little better. But I don’t have a strong opinion either way.

Reply:
My issue with the current writing is, if we ever were to have first items with align of 8 and then items with align of 16, then the alignment for the first of the items of the second type would be wrong. Because we align after `func1` with the alignment of `func1` and not of `func2`, to reuse your example.